Issue #168/string keys for consumer message headers

ColinDKelley opened the pull request:
Addresses issue #168:
- Change `Headers` to return `Hash<String, String>` as documented, starting with corresponding spec change.
- Rdoc updates.

These simplifications are also included:
- Change `Headers` to a `Module` since it is only a namespace and not a `Class`.
- Use `__dir__` for the directory of the current `__FILE__`.

Note: this is a breaking change since any code that is reading the `headers` today with symbols will need to change to strings. Because the version number starts with `0.`, Sem Ver doesn't require a major version bump. But we should advertise it broadly and at least do a minor version bump.

thijsc reviewed (changes requested):
Looks good! There's one spec failing.

geoff2k commented on spec/rdkafka/producer_spec.rb:
```diff
@@ -404,9 +404,9 @@ def call(_, handle)
-    expect(message.headers[:foo]).to eq "bar"
-    expect(message.headers[:baz]).to eq "foobar"
-    expect(message.headers[:foobar]).to be_nil
+    expect(message.headers['foo']).to eq "bar"
```
Nitpick: could we be consistent about using double quotes for strings?

ColinDKelley replied:
@geoff2k Fixed!

geoff2k reviewed (commented):
>  Because the version number starts with 0., Sem Ver doesn't require a major version bump. But we should advertise it broadly and at least do a minor version bump.

Is it perhaps worth discussing (maybe not in this PR), when a major version bump to rdkafka-ruby might be appropriate?

ColinDKelley commented:
@thijsc
- I fixed that spec.
- Also, I added a spec for `Headers` since that was missing: https://github.com/appsignal/rdkafka-ruby/pull/207/commits/0891e4d3d7c6db4d319a275613d378dcbf5c9f06
- And I was getting really nervous about breaking callers who might still expect Symbol keys. So I tried this pattern that raises an `ArgumentError` if they make that mistake. It's sort of the opposite of HashWIthIndifferentAccess. :) What do you think? https://github.com/appsignal/rdkafka-ruby/pull/207/commits/bfc9eaf763d52abb6e10bea243c67b39afd66fb9

ColinDKelley commented:
@thijsc
Here's a possible interim approach on Symbol-vs-String that is pretty close to `ActiveSupport::HashWithIndifferentAccess` but it's inline/dependency-free. It will tolerate a Symbol key (by converting it to String) but it emits a `Kern.warn` deprecation warning first. This will give developers some notice and a bit of time to switch over without breaking anything. (It documents that when version 1 ships, we will at that point break their code.) https://github.com/appsignal/rdkafka-ruby/pull/207/commits/ef94f01476553f94f2207706180a6c33c47d0351

ColinDKelley commented:
@thijsc Just following up to see if you have any thoughts on the above?

thijsc commented:
I think having backwards compatibility for a while would be nice.

ColinDKelley commented:
@thijsc Will do.

ColinDKelley commented:
@thijsc The build here is green after the rebase. 👍

mensfeld commented:
@thijsc there is one problem with this PR. Previous headers were not frozen. While it's not a big deal I want to point out, that this **is** an API change.
